### DISCUSSION TITLE
Initiating a ReadTheDocs

### DISCUSSION THREAD

**Discussion by amirmghaemi** [UMD-ARLIS/Graph-Benchmarking-Project#8] [Ideas]
As discussed on Tuesday, setting up a ReadTheDocs will require .rst files for different parts of our code. 

Here are my thoughts on how we should start this process:

1. Clean up the code
- Remove unnecessary files.
- I have made a new branch for us to start this process: [Cleanup Branch](https://github.com/UMD-ARLIS/Graph-Benchmarking-Project/tree/cleaned_branch)
2. Review the following:
- [reStructuredText](https://en.wikipedia.org/wiki/ReStructuredText): Format needed to make RTD pages
- [Sphinx](https://www.sphinx-doc.org/en/master/): We will use this for documentation generation
- [ReadTheDocs for Flagon Distill](https://incubator-flagon-distill.readthedocs.io/en/distill_toolkit_refactor/): This is a RTD page I setup for another project a while ago.
- [RST files for Distill](https://github.com/UMD-ARLIS/incubator-flagon-distill/tree/dev/docs): The files used to build the RTD page.
3. Setup a [ReadTheDocs](https://readthedocs.org/dashboard/) account. I suggest using your GitHub account for it.
4. Try this [reStructuredText Editor](https://www.tutorialspoint.com/online_restructure_editor.php).
5. We **may** need to make the repo public.

I welcome any other thoughts/ideas! Once we cleanup our code base, I can make the [Cleanup Branch](https://github.com/UMD-ARLIS/Graph-Benchmarking-Project/tree/cleaned_branch) be the master branch and initiate a RTD (import the project).

**Comment by osullik** [comment]
Registered for an account, also found a VSCode Extension for .rst files (haven't used it in a meaningful way yet) 

```
Name: reStructuredText
Id: lextudio.restructuredtext
Description: reStructuredText language support (RST/ReST linter, preview, IntelliSense and more)
Version: 189.3.0
Publisher: LeXtudio Inc.
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=lextudio.restructuredtext
```
